core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1898,8 +1928,31 @@ struct mobj_ffa *thread_spmc_populate_mobj_from_rx(uint64_t cookie)
 	if (!mf)
 		goto out;
 
-	if (set_pages(descr->address_range_array,
-		      READ_ONCE(descr->address_range_count), num_pages, mf)) {
+	num_regions = MIN(READ_ONCE(descr->address_range_count),
```

Comment:
We should check that `frag_len` is less than `length` and the size of the rx buffer.
I'm not sure we even need to look at `descr->address_range_count`.
``` C
offs += offsetof(struct ffa_mem_region, address_range_array);
range_count = (frag_len - offs) / sizeof(struct ffa_address_range);
```

Reply:
> We should check that `frag_len` is less than `length` and the size of the rx buffer. 
Done


> I'm not sure we even need to look at `descr->address_range_count`.
> 
> ```c
> offs += offsetof(struct ffa_mem_region, address_range_array);
> range_count = (frag_len - offs) / sizeof(struct ffa_address_range);
> ```
If I understand correctly, descr->address_range_count represents the number of constituent memory region descriptors.  And there is a formulation:
length - offs = sizeof (ffa_address_range) * descr->address_range_count
if descr->address_range_count does not satisfy this relation, it will eventually fail to pass the check as follows: 
```
	if (idx != num_pages) {
		mobj_ffa_spmc_delete(mf);
		goto out;
	}
```

Reply:
Sure, we could check `descr->address_range_count`, but depending on what we're doing with the value it might not be that useful.

Reply:
Do I need to do anything else? In addition, will the fix be merged into the optee 3.22 version？

Reply:
Please add Jens' review tag (below). I think it makes sense to take this for 3.22.0.

Reply:
Done